packages/cli/src/commands/deploy/spec.ts

```diff
@@ -0,0 +1,89 @@
+import { Question, Arg, Option, ArgsAndOpts } from '../../register-command';
+import { MethodArg } from '../../prompts/prompt';
+
+export const name = 'deploy';
+export const description = 'deploy a contract instance';
+
+export const args: Arg[] = [
+  {
+    name: 'contract',
+    async prompt(): Promise<Question> {
+      const choices = await import('../../prompts/choices');
+
+      // TODO: Make this include contracts from _unlinked_ dependencies.
+      const contracts = choices.contracts('all');
+
+      return {
+        message: 'Pick a contract to deploy',
+        choices: contracts,
+        validate: value => contracts.includes(value),
```

Comment:
Shouldn't we return a string with the error message, or have a `validationError` property?

Reply:
There is a default validation error message based on the parameter name, but it can probably be improved.

https://github.com/OpenZeppelin/openzeppelin-sdk/blob/a851d4ce1cc475e1b227c0decfe891b90b90a35a/packages/cli/src/register-command.ts#L167